flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/WindowedStream.java

```diff
@@ -121,6 +124,23 @@ public WindowedStream(KeyedStream<T, K> input,
 	}
 
 	/**
+	 * Sets the allowed lateness. If the {@link WindowAssigner} used
+	 * is in processing time, then the allowed lateness is set to 0.
+	 */
+	@PublicEvolving
+	public WindowedStream<T, K, W> setAllowedLateness(Time lateness) {
+		long millis = lateness.toMilliseconds();
+		if (allowedLateness < 0) {
+			throw new IllegalArgumentException("The allowed lateness cannot be negative.");
+		} else if (allowedLateness != 0 && !windowAssigner.isEventTime()) {
+			this.allowedLateness = 0;
```

Comment:
LOG is more natural. There is no reason to stop the job for the user (as it is not an error). We should just tell him that 
it does not make that much sense, so that he knows how to interpret the results he gets. Throwing an exception is somehow more drastic.